sycl/test-e2e/Graph/Threading/submit.cpp

```diff
@@ -4,15 +4,14 @@
 // Extra run to check for immediate-command-list in Level Zero
 // RUN: %if level_zero && linux %{env SYCL_PI_LEVEL_ZERO_USE_IMMEDIATE_COMMANDLISTS=1 %{l0_leak_check} %{run} %t.out 2>&1 | FileCheck %s --implicit-check-not=LEAK %}
 
-// Test submitting a graph in a threaded situation.
-// Note that we do not check the outputs because multiple concurrent executions
-// is indeterministic (and depends on the backend command management).
-// However, this test verifies that concurrent graph submissions do not trigger
-// errors nor memory leaks.
+// Test submitting a graph multiple times in a threaded situation.
+// According to spec: If graph is submitted multiple times, dependencies are automatically added by the runtime to prevent concurrent executions of an identical graph.
+// and so the result is deterministic and we can check the results.
 
 #include "../graph_common.hpp"
 
 #include <thread>
+constexpr unsigned SubmitsPerThread = 128;
```

Comment:
prefer to avoid using global variables, can move this into `main()` next to where `NumThreads` is defined.

Reply:
good point.  code refined.